tests/infrastructure/test_open_search_stack.py

```diff
@@ -0,0 +1,282 @@
+# Standard
+import pytest
+
+# Installed
+from aws_cdk.assertions import Match, Template
+
+# Local
+from sds_data_manager.stacks.opensearch_stack import OpenSearch
+
+
+@pytest.fixture(scope="module")
+def template(app, sds_id, env):
+    stack_name = f"opensearch-{sds_id}"
+    stack = OpenSearch(app, stack_name, sds_id, env=env)
+    template = Template.from_stack(stack)
+    return template
+
+
+def test_secrets_manager_resource_count(template):
+    template.resource_count_is("AWS::SecretsManager::Secret", 1)
+
+
+def test_secrets_manager_resource_properties(template):
+    template.has_resource(
+        "AWS::SecretsManager::Secret",
+        {
+            "DeletionPolicy": "Delete",
+            "UpdateReplacePolicy": "Delete",
+        },
+    )
+
+
+def test_iam_roles_resource_count(template):
+    template.resource_count_is("AWS::IAM::Role", 1)
+
+
+def test_expected_properties_for_iam_roles(template):
+    found_resources = template.find_resources(
```

Comment:
```suggestion
    found_resources = template.has_resource_properties(
```
since there is only one I think we can just use the assertion on resource properties?